src/data_structures/fmindex.rs

```diff
@@ -226,7 +275,7 @@ impl FMDIndex {
 
             // if size changed, add last interval to list
             if interval.size != forward_interval.size {
-                curr.push(interval);
+                curr.push(interval.clone());
```

Comment:
BiInterval should be copy, though, right? It's only a few numbers. And yes, the `.clone` is probably a relic of a time when BiInterval wasn't copy. Good catch.